### PR TITLE
[fix] `update --outdated` raises NonExistentKey with dev package

### PR DESCRIPTION
**Pull request by shimpeko**
closes https://github.com/pypa/pipenv/issues/5540

### The issue

https://github.com/pypa/pipenv/issues/5540

### The fix

Lookup the target package from the correct category instead of always using `[package]` category.

### The checklist

* [x] Associated issue
* [x] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix.rst`, `.feature.rst`, `.behavior.rst`, `.doc.rst`. `.vendor.rst`. or `.trivial.rst` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

**Comment by matteius** [comment]
@shimpeko to fix the news fragment add double backticks around `update --outdated`

File: news/5540.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix: `update --outdated` raises NonExistentKey with outdated dev packages
```
Comment (oz123) [inline news/5540.bugfix.rst:1]: The linter says:
```
rst ``code`` is two backticks............................................Failed
```
So this should be ``update --outdated``.

**Review by oz123 — CHANGES_REQUESTED** [review]

**Comment by oz123** [comment]
Thank you for your contribution! Please check the review.

File: news/5540.bugfix.rst (hunk 1 above)
Comment (shimpeko) [reply news/5540.bugfix.rst]: 👍 [3b5558b](https://github.com/pypa/pipenv/pull/5541/commits/3b5558b231457e571f89c7ff323f78ac00efcf94) fixed.

**Comment by shimpeko** [comment]
@oz123 thank you fore review 🙏

**Pull request merged**